src/Contact/Avatar.php

```diff
@@ -109,18 +109,32 @@ private static function storeAvatarCache(Image $image, string $filename, int $si
 
 		$filepath = DI::basePath() . $path;
 
-		$dirpath = dirname($filepath);
+		$dirpath = DI::basePath() . '/avatar/';
```

Comment:
Replaces lines 108-112
```suggestion
		$dirpath = DI::basePath() . '/avatar/';
		$filepath = $dirpath . $filename . $size . '.' . $image->getExt();
```

Reply:
Would be too easy :grin:

Reply:
`$path` is superfluous, it's only used in the `return` statement where it's computing returning the `$filepath` value again.

Reply:
We need the complete file path and the complete url. Also we need the file path that starts with the avatar folder (for the permission inheritance). So we need three different variables.

Reply:
The permission inheritance uses `$filename`, not `$path`, this variable is superfluous.

Reply:
But later we use `$path` as a return value.

Reply:
Yes but we could return `$filepath` instead which holds the same value.

Reply:
Ah, sorry, I just realized it's using `BaseUrl` instead of `BasePath` for the `$filepath`, please disregard.